### ISSUE TITLE
Image analysis fails after "building cache" and only prints "killed"

### ISSUE THREAD

**Issue by HanschDennis** [wagoodman/dive#389]
Hi, 
I'm not able to analyze a pretty big image (~30GB). The output is:
`Image Source: docker://myImage`  
`Fetching image... (this can take a while for large images)`  
`Analyzing image...`  
`Building cache...`  
`Killed`  

Is there an option to get more details why it fails? I have enough disk space, I think that should't be a problem.
Thank you.

**Comment by dsajanice** [comment]
+1

**Comment by HanschDennis** [comment]
@mark2185 No, unfortunately not. I also have not tried to reproduce it with another image.

**Comment by awsassets** [comment]
same problem as yours
dive a 4GB image and got self Killed

**Comment by emacdona** [comment]
I encountered this error today. ~45 GB image, unfortunately not public :-(

My machine has 64 GB of memory, and after reading your question regarding memory, I brought up Resource Monitor (I'm running docker in WSL2 on Windows 10) -- and watched it chew through all remaining memory.

Some context: this image contains an entire development environment with lots of tools installed. Its built using one big Dockerfile -- with a bunch of sequential "RUN"s. I wanted to use dive so I could see what each layer (RUN command) adds... in hopes that I could find a common base to be refactored into a multi stage build.

Which leads me to my question...

I suspect that finding a way to use less memory will be hard (if not impossible). In leiu of that, would you consider a feature that let one specify the "Last <n> layers" to be analyzed by dive? That would help my use case, at least :-)

I could optimize my HUGE docker image, say, 10 layers at a time.

Thanks for making dive!

**Comment by IssacKramer** [comment]
> I encountered this error today. ~45 GB image, unfortunately not public :-(
> 
> My machine has 64 GB of memory, and after reading your question regarding memory, I brought up Resource Monitor (I'm running docker in WSL2 on Windows 10) -- and watched it chew through all remaining memory.
> 
> Some context: this image contains an entire development environment with lots of tools installed. Its built using one big Dockerfile -- with a bunch of sequential "RUN"s. I wanted to use dive so I could see what each layer (RUN command) adds... in hopes that I could find a common base to be refactored into a multi stage build.
> 
> Which leads me to my question...
> 
> I suspect that finding a way to use less memory will be hard (if not impossible). In leiu of that, would you consider a feature that let one specify the "Last n layers" to be analyzed by dive? That would help my use case, at least :-)
> 
> I could optimize my HUGE docker image, say, 10 layers at a time.
> 
> Thanks for making dive!

+1 same here. 
Cannot analyze image based on windows Server Core (5G) - telling me :
" path error at layer index Index(0-47:48): unable to add ......"
So maybe a feature to analyze only X top numbers of layers (I won't optimize microsoft layers.....)
Thx.
